pytorch_lightning/trainer/trainer.py

```diff
@@ -894,6 +891,28 @@ def enable_validation(self) -> bool:
         val_loop_enabled = (self.is_overridden('validation_step') and self.limit_val_batches > 0)
         return val_loop_enabled or self.fast_dev_run
 
+    @property
+    def default_root_dir(self) -> str:
+        """
+        The default location to save artifacts of loggers, checkpoints etc.
+        It is used as a fallback if logger or checkpoint callback do not define specific save paths.
+        Defaults to ``os.getcwd()``.
+        """
+        path = self._default_root_dir or os.getcwd()
+        path = os.path.normpath(path)
+        return path
+
+    @property
+    def weights_save_path(self) -> str:
+        """
+        The default root location to save weights (checkpoints), e.g., when the
+        :class:`~pytorch_lightning.callbacks.model_checkpoint.ModelCheckpoint` does not define a file path.
+        It defaults to :paramref:`~pytorch_lightning.trainer.trainer.Trainer.default_root_dir`.
+        """
+        path = self._weights_save_path or self.default_root_dir
```

Comment:
this can be tricky as default mind be `os.getcwd()`

Reply:
why? this is exactly what we want, and what was the default before. I did not change it.

Reply:
moved to init now as you suggested